omnigibson/utils/motion_planning_utils.py

```diff
@@ -544,3 +545,76 @@ def detect_robot_collision_in_sim(robot, filter_objs=[], ignore_obj_in_hand=True
         if col_obj.category in filter_categories:
             collision_prims.remove(col_prim)
     return len(collision_prims) > 0
+
+def astar(search_map, start, goal, eight_connected=True):
```

Comment:
this is awesome! one question: what is the slowdown here (running loop in python vs. c++)? There's some other open-source python packages that apparently implement A* in C++ (such as [this one](https://github.com/hjweide/pyastar2d)), so I'm curious if there was a reason for the custom implementation here.

I'm by no means an expert, so I'll defer to you. Also cc'ing @cgokmen if he has any further insight too

Reply:
we aren't slowing this down relative to what's already happening iiuc. Previously this was being done in networkx which concretely is slower than this implementation because it is in python and can't benefit from the domain specific knowledge of the 2d grid.

you are right that there are c++ implementations that could be used directly for this purpose. I am however hesitant to add those as a dependency - they all appear unmaintained with known issues. if this becomes an issue later we can take a look.

Reply:
I was hesitant to use any external library for this cuz we would be essentially adding another dependency after removing networkx. I'm slightly inclined to keep it like this for now. If this causes any performance issue, it should be super simple to swap this implementation out. If we can reuse this A* implementation in any other part of our code base, that would also be great (I can try to make this more generic to fit more use cases if needed, e.g. support for higher dimension).